packages/editor/CodeMirror/createEditor.ts

```diff
@@ -0,0 +1,297 @@
+		if (!viewUpdate.state.selection.eq(viewUpdate.startState.selection)) {
+			const mainRange = viewUpdate.state.selection.main;
+			const event: SelectionRangeChangeEvent = {
+				kind: EditorEventType.SelectionRangeChange,
+
+				anchor: mainRange.anchor,
+				head: mainRange.head,
+				from: mainRange.from,
+				to: mainRange.to,
+			};
+			props.onEvent(event);
+		}
+	};
+
+	const notifySelectionFormattingChange = (viewUpdate?: ViewUpdate) => {
+		const spellcheck = globalSpellcheckEnabled();
+
+		// If we can't determine the previous formatting, post the update regardless
+		if (!viewUpdate) {
+			const formatting = computeSelectionFormatting(editor.state, spellcheck);
+			props.onEvent({
+				kind: EditorEventType.SelectionFormattingChange,
+				formatting,
+			});
+		} else if (viewUpdate.docChanged || !viewUpdate.state.selection.eq(viewUpdate.startState.selection)) {
+
+			// Only post the update if something changed
+			const oldFormatting = computeSelectionFormatting(viewUpdate.startState, spellcheck);
+			const newFormatting = computeSelectionFormatting(viewUpdate.state, spellcheck);
+
+			if (!selectionFormattingEqual(oldFormatting, newFormatting)) {
+				props.onEvent({
+					kind: EditorEventType.SelectionFormattingChange,
+					formatting: newFormatting,
+				});
+			}
+		}
+	};
+
+	// Returns a keyboard command that returns true (so accepts the keybind)
+	// alwaysActive: true if this command should be registered even if ignoreModifiers is given.
+	const keyCommand = (key: string, run: Command, alwaysActive?: boolean): KeyBinding => {
+		return {
+			key,
+			run: editor => {
+				if (settings.ignoreModifiers && !alwaysActive) return false;
+
+				return run(editor);
+			},
+		};
+	};
+
+	const dynamicConfig = new Compartment();
+
+	const editor = new EditorView({
+		state: EditorState.create({
+			// See https://github.com/codemirror/basic-setup/blob/main/src/codemirror.ts
+			// for a sample configuration.
+			extensions: [
+				dynamicConfig.of([
+					configFromSettings(props.settings),
+				]),
+				history(),
+				search(settings.useExternalSearch ? {
+					createPanel(_: EditorView) {
+						return {
+							// The actual search dialog is implemented with react native,
+							// use a dummy element.
+							dom: document.createElement('div'),
+							mount() {
+								showSearchDialog();
+							},
+							destroy() {
+								hideSearchDialog();
+							},
+						};
+					},
+				} : undefined),
+				drawSelection(),
+				highlightSpecialChars(),
+				highlightSelectionMatches(),
+				indentOnInput(),
+
+				EditorView.domEventHandlers({
+					scroll: (_event, view) => {
+						props.onEvent({
+							kind: EditorEventType.Scroll,
+							fraction: getScrollFraction(view),
+						});
+					},
+				}),
+
+				// TODO: indent with tabs to match other editors.
+				// Tab indentation has list-related bugs. See
+				// https://github.com/codemirror/dev/issues/1243
+				indentUnit.of('    '),
+				EditorState.tabSize.of(4),
```

Comment:
This has been fixed upstream. However, the mobile app currently indents with spaces.

A follow-up pull request can change this to `indentUnit.of('\t')` for consistency with CodeMirror 5, but should add a setting to use space indentation on mobile.